exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -893,6 +893,87 @@ func TestReconcileMachinePoolInfrastructure(t *testing.T) {
 				g.Expect(m.Status.InfrastructureReady).To(BeFalse())
 			},
 		},
+		{
```

Comment:
Could we add additional scale to zero, scale from zero tests in this file?

Reply:
I'm also curious what happens if you scale to zero and try to scale back up. It would be good to double check that it doesn't enter an invalid state.

Reply:
thanks for the idea - I added another test (see below in the file) which tests the scale up & down scenarios. 
Indeed I catched a bug with this and had to adjust the noderef reconciler slightly. When scaling to zero the last node also needs to be deleted, and one condition there prevented that.